pyproject.toml

```diff
@@ -143,12 +146,26 @@ junit_logging = "all"
 asyncio_mode = "strict"
 xfail_strict = true
 
-# With rare exceptions, Rally does not use type hints. The intention of the
-# following largely reduced mypy configuration scope is verification of argument
-# types in config.Config methods while introducing configuration properties
-# (props). The error we are after here is "arg-type".
+# With exceptions specified in mypy override section, Rally does not use type
+# hints (they were a novelty when Rally came to be). Hints are being slowly and
+# opportunistically introduced whenever we revisit a group of modules.
+#
+# The intention of the following largely reduced global config scope is
+# verification of argument types in config.Config methods while introducing
+# configuration properties (props). The intention of "disable_error_code" option
+# is to keep "arg-type" error code, while disabling other error codes.
+# Ref: https://github.com/elastic/rally/pull/1798
 [tool.mypy]
-python_version = 3.8
+python_executable="$VIRTUAL_ENV/bin/python"
```

Comment:
@gbanasiak, can you test `make lint` with this line? I'm hoping this works everywhere.

It seems to work properly with both my local setup and the CI so far.